### PR TITLE
Increase username limit to 24 and ensure usernames are valiated properly

### PR DESCRIPTION
**Pull request by cbreezier**

### CODE REVIEW

File: modules/users/client/config/users.client.config.js [hunk 1]
```diff
@@ -28,10 +28,3 @@ angular.module('users').config(['$httpProvider',
     ]);
   }
 ]);
-
-angular.module('users').constant('settings', {
```
Comment (cbreezier) [inline modules/users/client/config/users.client.config.js:32]: Pretty sure these are no longer used now that I've deleted the frontend angular directive (which was never used).

File: modules/users/server/controllers/users/users.authentication.server.controller.js [hunk 2]
```diff
@@ -20,11 +20,22 @@ var noReturnUrls = [
...
  */
 exports.signin = function (req, res, next) {
-  passport.authenticate('local', function (err, user, info) {
+  passport.authenticate('local', async function (err, user, info) {
     if (!user) {
       user = new User(req.body);
       user.provider = 'local';
-      user.save();
+      try {
+        await user.save();
```
Comment (cbreezier) [inline modules/users/server/controllers/users/users.authentication.server.controller.js:28]: This is the main good fix - make sure we care about the result of saving the user.

File: modules/users/server/controllers/users/users.authentication.server.controller.js [hunk 3]
```diff
@@ -20,11 +20,22 @@ var noReturnUrls = [
...
+      } catch (e) {
+        if (e instanceof mongoose.Error.ValidationError) {
```
Comment (cbreezier) [inline modules/users/server/controllers/users/users.authentication.server.controller.js:30]: This was difficult to find. I had to url-hack the docs to find old v4.x mongoose docs.

File: modules/users/server/controllers/users/users.authentication.server.controller.js [hunk 4]
```diff
@@ -20,11 +20,22 @@ var noReturnUrls = [
...
+          res.status(400).send(e);
+          return;
+        } else {
+          logger.warn('Failed to create new user %s: %s', req.body.username, e.toString());
```
Comment (cbreezier) [inline modules/users/server/controllers/users/users.authentication.server.controller.js:34]: Dunno how to log more complex objects. We use "winston" which seems pretty simple...

**Review by delete12 — APPROVED** [review]

**Pull request merged**